observatory-dags/observatory/dags/telescopes/google_analytics.py

```diff
@@ -0,0 +1,515 @@
+from observatory.platform.telescopes.snapshot_telescope import SnapshotRelease, SnapshotTelescope
+from observatory.platform.utils.airflow_utils import AirflowConns, AirflowVars
+from observatory.platform.utils.telescope_utils import list_to_jsonl_gz, make_dag_id
+from observatory.platform.utils.template_utils import blob_name, bq_load_shard_v2, table_ids_from_path
+
+
+class GoogleAnalyticsRelease(SnapshotRelease):
+    def __init__(self, dag_id: str, start_date: pendulum.Pendulum, release_date: pendulum.Pendulum,
+                 organisation: Organisation):
+        """ Construct a GoogleAnalyticsRelease.
+
+        :param dag_id: the id of the DAG.
+        :param start_date: the start date of the download period.
+        :param release_date: the end date of the download period, also used as release date for BigQuery table and
+        file paths
+        :param organisation: the Organisation of which data is processed.
+        """
+        self.dag_id_prefix = GoogleAnalyticsTelescope.DAG_ID_PREFIX
+        transform_files_regex = f"{self.dag_id_prefix}.jsonl.gz"
+
+        super().__init__(dag_id=dag_id, release_date=release_date,
+                         transform_files_regex=transform_files_regex)
+
+        self.organisation = organisation
+        self.start_date = start_date
+        self.end_date = release_date
+
+    @property
+    def download_bucket(self):
+        """ The download bucket name.
+        :return: the download bucket name.
+        """
+        return self.organisation.gcp_download_bucket
+
+    @property
+    def transform_bucket(self):
+        """ The transform bucket name.
+        :return: the transform bucket name.
+        """
+        return self.organisation.gcp_transform_bucket
+
+    @property
+    def transform_path(self) -> str:
+        """ Get the path to the transformed file.
+
+        :return: the file path.
+        """
+        return os.path.join(self.transform_folder, f'{self.dag_id_prefix}.jsonl.gz')
+
+    def download_transform(self, view_id: str, pagepath_regex: str) -> bool:
+        """ Downloads and transforms an individual Google Analytics release.
+
+        :param view_id: The view id.
+        :param pagepath_regex: The regex expression for the pagepath of a book.
+        :return: True when data available for period, False if no data is available
+        """
+
+        service = initialize_analyticsreporting()
+        results = get_reports(service, self.organisation.name, view_id, pagepath_regex, self.start_date, self.end_date)
+        if results:
+            list_to_jsonl_gz(self.transform_path, results)
+            return True
+        else:
+            return False
+
+
+class GoogleAnalyticsTelescope(SnapshotTelescope):
+    """ Google Analytics Telescope."""
+    DAG_ID_PREFIX = 'google_analytics'
+
+    def __init__(self, organisation: Organisation, extra: dict, dag_id: Optional[str] = None,
+                 start_date: datetime = datetime(2021, 1, 1),
```

Comment:
I would prefer to have view_id and pagepath_regex as explicit parameters in the constructor (rather than passing the extra dictionary), because it makes the GoogleAnalyticsTelescope constructor interface clearer.